lib/positioning/healer.rb

```diff
@@ -8,8 +8,8 @@ def initialize(model, column, order)
 
     def heal
       if positioning_columns.present?
-        @model.select(*positioning_columns).distinct.each do |scope_record|
-          sequence @model.where(scope_record.slice(*positioning_columns))
+        @model.unscope(:select, :order).select(*positioning_columns).distinct.each do |scope_record|
+          sequence @model.unscope(:select).where(scope_record.slice(*positioning_columns))
```

Comment:
@brendon I changed the test case to include not only the `order` clause but also a `select` clause in the default scope. The tests only pass if I also undo the `select` on line 12 (in addition to 11), but I am unsure why this is needed.
If I understand the code correctly, the `#sequence`  method takes a scope, overrides the ordering, and updates the position attribute. This should be independent of any select clause in the default scope, no?

Reply:
That's a bit weird eh?! There won't even be a default scope of `select` to unscope. Did you try `reselect` instead of adding `:select` to the first `unscope` call?